### PR TITLE
Add config flow to IMAP integration

### PR DESCRIPTION
**Pull request by engrbm87**
## Proposed change

Update documentation for IMAP integration now supporting config flow.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:  https://github.com/home-assistant/core/pull/74623
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/imap.markdown [hunk 1]
```diff
@@ -84,27 +37,24 @@ By default, this integration will count unread emails. By configuring the search
...
+
+# Example entry for Office 365
+
+  server: outlook.office365.com
+  port: 993
+  username: email@address.com
+  password: password
+  search: FROM <sender@email.com>, SUBJECT <subject here>
+  charset: US-ASCII
 ```
```
Comment (engrbm87) [inline source/_integrations/imap.markdown]: Appreciate the help on how to highlight these as config entry examples.

File: source/_integrations/imap.markdown [hunk 2]
```diff
@@ -9,65 +9,18 @@ ha_domain: imap
 ha_platforms:
   - sensor
 ha_integration_type: integration
+ha_codeowners:
+  - '@engrbm87'
+ha_config_flow: true
 ---
 
-The `imap` integration is observing your [IMAP server](https://en.wikipedia.org/wiki/Internet_Message_Access_Protocol) and reporting the amount of unread emails.
+The `imap` integration is observing your [IMAP server](https://en.wikipedia.org/wiki/Internet_Message_Access_Protocol) and reporting the amount of unread emails. Other search criteria can be used as shown in the example below.
```
Comment (frenck) [inline source/_integrations/imap.markdown]:
```suggestion
The IMAP integration is observing your [IMAP server](https://en.wikipedia.org/wiki/Internet_Message_Access_Protocol) and reporting the number of unread emails. Other search criteria can be used, as shown in the example below.
```

File: source/_integrations/imap.markdown [hunk 3]
```diff
@@ -84,27 +37,24 @@ By default, this integration will count unread emails. By configuring the search
...
-sensor:
-  - platform: imap
-    server: imap.gmail.com
-    port: 993
-    username: YOUR_USERNAME
-    password: YOUR_PASSWORD
-    search: FROM <sender@email.com>, SUBJECT <subject here>
-    # Or use X-GM-RAW search-term like this, to find unread emails from the last 7 days in your inbox
-    # search: 'X-GM-RAW "in: inbox newer_than:7d is:unread"'
+#### Config entry sample with search examples
```
Comment (frenck) [inline source/_integrations/imap.markdown]: We do not use the "Config entry" term much in user-facing documentation or UIs. 
A user won't have an idea what this headline means.

File: source/_integrations/imap.markdown [hunk 4]
```diff
@@ -84,27 +37,24 @@ By default, this integration will count unread emails. By configuring the search
...
-# Example configuration.yaml entry for Office 365
-sensor:
-  - platform: imap
-    server: outlook.office365.com
-    port: 993
-    username: email@address.com
-    password: password
-    search: FROM <sender@email.com>, SUBJECT <subject here>
-    charset: US-ASCII
+```
```
Comment (frenck) [inline source/_integrations/imap.markdown]:
````suggestion
```txt
````

File: source/_integrations/imap.markdown [hunk 5]
```diff
@@ -84,27 +37,24 @@ By default, this integration will count unread emails. By configuring the search
...
+  # search: 'X-GM-RAW "in: inbox newer_than:7d is:unread"'
+
+# Example entry for Office 365
+
+  server: outlook.office365.com
+  port: 993
+  username: email@address.com
+  password: password
+  search: FROM <sender@email.com>, SUBJECT <subject here>
+  charset: US-ASCII
```
Comment (frenck) [inline source/_integrations/imap.markdown]: What would a user do with these? Is it YAML? Do we paste this in a big text field? This is not clear from the provided text.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/imap.markdown (hunk 5 above)
Comment (engrbm87) [reply source/_integrations/imap.markdown]: These are examples for the user input while creating a new entry. That is what I need help showing in the documentation.

Comment (frenck) [reply source/_integrations/imap.markdown]: Just write it out as introduction text, with a list of labels with values.

Something like:

---
Let's say you want to connect your Office account, with a specific search thingy. You could use the following values during the integration setup:

- **Server**: `outlook.office365.com`
- **Port**: `993`
- **Username**: __Your full email address__
- **Password**: __Your password__
- **Search**: `FROM <sender@email.com>, SUBJECT <subject here>`
- **Charset**: `US-ASCII`

Comment (engrbm87) [reply source/_integrations/imap.markdown]: Thanks, please check the updated documentation.

**Review by frenck — APPROVED** [review]
Thanks, @engrbm87 👍

**Pull request merged**